lib/improver/weighted_blend.py

```diff
@@ -33,13 +33,208 @@
 
 import numpy as np
 import iris
+from iris.analysis import Aggregator
 
 
-class BasicWeightedAverage(object):
-    """Apply a Basic Weighted Average to a cube."""
+class BlendingUtilities(object):
+    """Class for blending utilities functions"""
+
+    def __init__(self):
+        """
+        Initialise class.
+        """
+        pass
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<BlendingUtilities>')
+        return result
+
+    @staticmethod
+    def basic_weighted_average(cube, weights, coord, coord_dim):
+        """Calculate weighted mean across the chosen coord
+
+        Args:
+            cube : iris.cube.Cube
+                   Cube to blend across the coord.
+            weights: Optional list or np.array of weights
+                     or None (equivalent to equal weights)
+            coord : string
+                     The name of a coordinate dimension in the cube.
+            coord_dim : tuple
+                     The index of the coordinate dimension in the cube.
+
+        Returns:
+            result : iris.cube.Cube
+                     containing the weighted mean across the chosen coord
+        """
+        # Supply weights as an array of weights whose shape matches the cube.
+        weights_array = None
+        if weights is not None:
+            weights_array = iris.util.broadcast_to_shape(np.array(weights),
+                                                         cube.shape,
+                                                         coord_dim)
+        # Calculate the weighted average.
+        result = cube.collapsed(coord,
+                                iris.analysis.MEAN, weights=weights_array)
+        return result
+
+    @staticmethod
+    def blend_percentile_cube(cube, weights, coord, coord_dim, perc_coord):
+        """ Blend together percentile cube
+
+        Args:
+            cube : iris.cube.Cube
+                   Cube to blend across the coord.
+            weights: Optional list or np.array of weights
+                     or None (equivalent to equal weights)
+            coord : string
+                     The name of a coordinate dimension in the cube.
+            coord_dim : tuple
+                     The index of the coordinate dimension in the cube.
+            perc_coord : iris.cube.DimCoord
+                     The perecentile coordinate
+
+        Returns:
+            result : iris.cube.Cube
+                     containing the weighted percentile blend
+                     across the chosen coord
+        """
+
+        percentiles = np.array(perc_coord.points, dtype=float)
```

Comment:
I think I've got rid of this line, but it might be needed to make sure the percentile values are floats